### DISCUSSION TITLE
raw query inside transaction

### DISCUSSION THREAD

**Discussion by namhihi237** [prisma/prisma#13308] [Q&A]
Hi all,
I tried using raw query inside transaction but have an error.
 ```
  const [lastIncrement] = await ctx.prisma.$transaction([
    ctx.prisma.$queryRaw(`SELECT counter FROM increment_counter WHERE code = 'INV-${yearCode}' FOR UPDATE;`),
    ctx.prisma.$executeRaw(`UPDATE increment_counter SET counter = counter + 1;`)
  ]);
```

Error: "Raw query failed. Code: `1205`. Message: `Lock wait timeout exceeded; try restarting transaction`\"

Prisma Version: 2.17.0
Window 10
Node version: 14.17.0

**Comment by nurul3101** [comment]
Hey @namhihi237 👋 
Which database are you using?
Could you try upgrading to the latest version of prisma, we have made substantial improvements in the newer releases?

**Reply by vladi-stevanovic** [reply]
👋 Hello @namhihi237  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
